File: client/src/types/post-tag.ts
export interface PostTagResponse {
    id: string
    name: string
    slug: string
    color?: string
    count: number
}

export interface PostTagCreate {
    name: string
    slug: string
    color?: string
}

export interface PostTagUpdate {
    name?: string
    slug?: string
    color?: string
}

File: archive/server-v0/app/schemas/post_tag.py
from pydantic import BaseModel


class PostTagCreate(BaseModel):
    name: str
    slug: str
    color: str | None = None


class PostTagUpdate(BaseModel):
    name: str | None = None
    slug: str | None = None
    color: str | None = None


class PostTagResponse(BaseModel):
    id: str
    count: int = 0

    name: str
    slug: str
    color: str | None = None

    model_config = {"from_attributes": True}
